src/client/connect/http.rs

```diff
@@ -256,7 +274,7 @@ where
     R: Resolve + Clone + Send + Sync + 'static,
     R::Future: Send,
 {
-    type Response = TcpStream;
+    type Response = HttpConnection;
```

Comment:
This would be a breaking change. Can we provide the configuration without needing a new type?